tribe_client/utils.py

```diff
@@ -407,17 +408,23 @@ def download_organism_public_genesets(organism, creator_username=None):
     downloaded from Tribe to get only the ones created by the user with
     this username.
 
+    request_params -- Optional argument, a dictionary of parameters to be
+    sent with the request to get gene sets from Tribe
+    (e.g. 'full_annotations').
+
     Returns:
     all_public_genesets -- A dictionary of GO, KEGG and DO terms available
     in Tribe for the specified organism (and optionally, creator_username).
     """
+    request_params['organism__scientific_name'] = organism
+
+    if 'show_tip' not in request_params or not request_params['show_tip']:
+        request_params['show_tip'] = 'true'
 
     # *Note: Tribe does not like requests for more than 1500
     # genesets at a time, so use this as the 'limit' parameter.
-    request_params = {
-        'show_tip': 'true', 'limit': '1500',
-        'organism__scientific_name': organism
-    }
+    if 'limit' not in request_params or not request_params['limit']:
```

Comment:
What if `request_params['limit']` is 0? Does it make sense to reset it to 1500?

Reply:
I think I see why. So both `request_params['show_tip']` and `request_params['show_tip']` must be string type, am I right?

Reply:
Ahh good question about the `'limit'` being `0`. It probably would be good to reset it to 1500, because otherwise, they will probably get a timeout error from Tribe (because of the huge amount of genesets). We should probably also throw a warning to let users know that the limit is being reset.

Reply:
I think both `'show_tip'` and `'limit'` need to be strings, but let me check.

Reply:
If both `request_params['show_tip']` and `request_params['limit']` are strings, then unless they are empty string:
`len(request_params['show_tip']) == 0`
or:
`len(request_params['limit']) == 0`
the default value won't be set. This will work for `request_params['show_tip']`, but for `request_params['limit']`, it will never reset any meaningful value (including "0") to 1500. So maybe the logic should be:
```
if 'limit' not in request_params or not request_params['limit'] or int(request_params['limit']) == 0 or int(request_params['limit'] > 1500:
        request_params['limit'] = '1500'

Reply:
Or code like this:
```
request_params['limit'] = '1500'
try:
    limit_value = int(request_params['limit']);
    if limit_value > 0 and limit_value < 1500:
        request_params['limit'] = limit_value
except:  // in case 'limit' doesn't exist or not an integer
    pass
```
I wonder whether it can be better.

Reply:
So I checked and actually `'show_tip'` should *definitely* be a string, but `'limit'` can be either a `string` or an `int`.

Reply:
The code I wrote earlier has a serious bug. So I have deleted it. Here is the corrected version:
```
limit_value = 1500
try:
    req_value = int(request_params['limit']);
    if req_value > 0 and req_value < 1500:
        limit_value = req_value
except:  
    # put a  warning message here maybe ...

request_params['limit'] = limit_value
```

Reply:
except maybe without the JavaScript semicolon at the end of the second line ;p

Reply:
Oh, yeah. I blame this mostly to C++.